### PR TITLE
Systick example

### PR DESCRIPTION
**Pull request by hmightypirate**

### CODE REVIEW

File: systick_src/hal/clock.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+uint32_t get_counter(void) {
+  return counter;
+}
+
+/*
+ *  @brief systick interruption routine. It counts from 0 to 10000
+ */
+void sys_tick_handler(void) {
+
+  // FIXME: app dependent function
```
Comment (eLeDe) [inline systick_src/hal/clock.c:17]: Remove this comment

File: systick_src/main.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#include "hal/setup.h"
+#include "hal/discovery_leds.h"
+#include "hal/clock.h"
+
+void wait(void);
```
Comment (eLeDe) [inline systick_src/main.c]: Wait function with nops is not needed anymore

File: systick_src/main.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+{
+
+  setup();
+  
+  uint32_t consolidated_counter = 0;
+
+  systick_interrupt_enable();
+  
+  while(1) {
+    wait();
```
Comment (eLeDe) [inline systick_src/main.c]: Remove wait function

**Review by eLeDe — CHANGES_REQUESTED** [review]
Wait function is not needed

**Comment by hmightypirate** [comment]
Done

**Pull request merged**